### DISCUSSION TITLE
Some improvements to UI & widgets

### DISCUSSION THREAD

**Discussion by shuvashish76** [LibreShift/red-moon#300] [Ideas]
https://user-images.githubusercontent.com/48226312/119297212-26eb8a00-bc78-11eb-81ea-02cb4334031f.mp4

(everything tested on the video mentioned below chronologically)
1. while moving the app on the home screen there is a white background behind the Red Moon logo. Just like other apps, it would look great if the "wolf" moves between the moon only.
2. when we toggle 2nd widget that is Red Moon Toggle there is a Red flash screen (check force stop app & toggle 2nd widget again) [probably android limitations, not a big issue]
>btw I love Red Moon "Lower system brightness" feature which is missing in [Night Screen](https://github.com/fython/Blackbulb) app. 💖

![new moon widget](https://user-images.githubusercontent.com/48226312/119297561-d9235180-bc78-11eb-8359-23207617f86e.jpeg)
3. Currently there are two widgets but both do the same job (2nd widget is for toggle on/off & 1st widget show its status only)
_few suggestions_
I) 1st widget looks a bit ugly now & I don't think a widget only for status purpose is worth
Make 1st widget a circular transparent wolf logo just like [Loop Habit tracker](https://github.com/iSoron/uhabits) app widget (check video) rename it to "Toggle Red Moon". & it should do the job of _current_ 2nd widget that is toggle on/off the Red Moon
![Screenshot_2021-05-26-18-04-56](https://user-images.githubusercontent.com/48226312/119660637-1cceb480-be4d-11eb-92e0-541f455408c0.jpeg)
![Screenshot_2021-05-26-18-05-21](https://user-images.githubusercontent.com/48226312/119660693-28ba7680-be4d-11eb-8460-81fd6c4fa523.jpeg)

II) & we should rename the 2nd widget to "Red Moon" only & clicking it should open up notification slider & should do the job of controlling Darkness, Colour & Opacity from system notification slider just like Night Screen app as mentioned in #294

**Comment by smichel17** [comment]
This is a bunch of things, so I'm going to convert it to a discussion.

**Comment by smichel17** [comment]
> 2. when we toggle 2nd widget that is Red Moon Toggle there is a Red flash screen (check force stop app & toggle 2nd widget again) [probably android limitations, not a big issue]

This is because the 2nd "widget", with the Red Moon icon is not really a widget, it's a shortcut. See https://github.com/LibreShift/red-moon/issues/132#issuecomment-287654952.

Ideally, we could totally hide the shortcut "widget". However, I don't know if this is possible.  
The next best thing would be to nudge people to use the other widget. I think we can do that by making the right widget prettier, and the shortcut uglier— added that comment to #201. I think that would also address:

> 1. while moving the app on the home screen there is a white background behind the Red Moon logo. Just like other apps, it would look great if the "wolf" moves between the moon only

I'm not sure how much your other widget UI ideas change with #201

**Reply by shuvashish76** [reply]
I see. Thanks for the detailed explanation.

![Screenshot_2021-09-13-02-24-57](https://user-images.githubusercontent.com/48226312/133002608-2c9a20ee-6253-46bf-8175-563e7b76af7d.jpeg)
But both 1. _Red Moon_ (real widget) & 2. _Red Moon Toggle_ (shortcut) comes under "Widgets" that leads to an assumption that both are widgets...

We can separate the shortcut one from the widget to something like this👇
![Screenshot_2021-09-13-02-19-45](https://user-images.githubusercontent.com/48226312/133002698-f6118d6f-1270-40fb-8a31-a4acf2ef5d27.jpeg)
(shortcut options are marked with red box & widget option is marked with yellow box)

how shortcuts looks👇
![Screenshot_2021-09-13-02-36-29](https://user-images.githubusercontent.com/48226312/133002809-b8bf4ae8-7ad5-4ca2-b29a-760a4a35d559.jpeg)

Screenshots from [Quotes Status Creator](https://github.com/VishnuSanal/Quotes) app

**Reply by shuvashish76** [reply]
But again if we add multiple widgets & Toggling on/off as a Widget (as you mentioned [here](https://github.com/LibreShift/red-moon/discussions/300#discussioncomment-1303724) then IMO we can _remove_ these _shortcut features completely_...

[Loop Habit Tracker](https://github.com/iSoron/uhabits) has multiple widgets but **no shortcuts**. Something like that is definitely possible...

**Reply by smichel17** [reply]
The primary purpose of the shortcut is to allow people to toggle the filter through Tasker/Easer or similar apps, without those apps having to build in support for Red Moon specifically. If that's possible some other way, then we can get rid of it.

Someone else would need to do the research on what's needed for that to happen, though, as it's not really an important feature [**for me**](https://github.com/LibreShift/red-moon/issues/281#issue-771300115).

**Comment by smichel17** [comment]
> 1st widget show its status only

That widget should also toggle Red Moon. If it's not working, that's a bug.

**Reply by shuvashish76** [reply]
I see then it is a bug. It really never worked for me.

**Comment by smichel17** [comment]
> we should rename the 2nd widget to "Red Moon" only & clicking it should open up notification slider & should do the job of controlling Darkness, Colour & Opacity from system notification slider just like Night Screen app as mentioned in #294

I'd like to make tapping the notification open this view, similar to Night Screen, which is already tracked in #294 as you mentioned :). We could also add another widget which opens it, but I _think_ there should still be a widget for toggling (since the shortcut "widget" is not really intended to be used).

**Reply by shuvashish76** [reply]
I like the idea of widget for Floating UI (#294). & I agree that Toggling on/off Widget is necessary.👍
